anilist.py

```diff
@@ -172,54 +172,78 @@ def match_to_plex(anilist_series: List[AnilistSeries], plex_series_watched: List
         plex_seasons = plex_series.seasons
         plex_anilist_id = plex_series.anilist_id
 
-        custom_mapping_seasons_anilist_id = 0
-        mapped_season_count = 0
-        plex_watched_episode_count_custom_mapping = 0
+        custom_mapped_seasons = []
 
         logger.info("--------------------------------------------------")
 
         # Check if we have custom mappings for all seasons (One Piece for example)
         if len(plex_seasons) > 1:
-            custom_mapping_season_count = 0
+            anilist_matches = []
             for index, plex_season in enumerate(plex_seasons):
+
                 season_mappings = retrieve_season_mappings(
                     plex_title, plex_season.season_number
                 )
-                matched_id = 0
                 if season_mappings:
                     matched_id = season_mappings[0].anime_id
-                    if custom_mapping_seasons_anilist_id in (0, matched_id):
-                        custom_mapping_season_count += 1
-                        custom_mapping_seasons_anilist_id = matched_id
-
-                        # Check if season should be added together
-                        if (index != 0 and plex_season.first_episode > plex_seasons[index - 1].last_episode):
-                            plex_watched_episode_count_custom_mapping = plex_season.watched_episodes
-                        else:
-                            plex_watched_episode_count_custom_mapping += plex_season.watched_episodes
-
-            # If we had custom mappings for multiple seasons with the same ID use
-            # cumulative episode count and skip per season processing
-            if custom_mapping_season_count > 1:
-                logger.warning(
-                    "[ANILIST] Found same custom mapping id for multiple seasons "
-                    "so not using per season processing but updating as one | "
-                    f"title: {plex_title} | anilist id: {custom_mapping_seasons_anilist_id} | "
-                    f"total watched episodes: {plex_watched_episode_count_custom_mapping}"
+
+                    custom_mapped_seasons.append(plex_season.season_number)
+                    match = next(
+                        (item for item in anilist_matches if item['anilist_id'] == matched_id),
+                        None,
+                    )
+
+                    if not match:
+                        anilist_matches.append({
+                            "anilist_id": matched_id,
+                            "watched_episodes": plex_season.watched_episodes,
+                            "total_episodes": plex_season.last_episode,
```

Comment:
total_episodes is never used and incremented incorrectly for the `elif (indes != 0` case

Reply:
Should have asked this. 
Should we allow the users to update a show on anilist if they don't have all the episodes on the previous season watched?
Like the example in the issue?

Should we create a configuration that always updates the anilist item with `watched episodes this season` + `number of episodes in previous seasons`? That was my intention with this last_episode parameter.

```
So with this
seasons:
      - season: 1
        anilist-id: 6702
        total_episodes: 25
        watched_episodes: 25
      - season: 2
        anilist-id: 6702
        total_episodes: 50
        watched_episodes: 10
      - season: 3
        anilist-id: 6702
        total_episodes: 50
        watched_episodes: 10
```

The calculations for episodes watched would be different. With the current setup we would always update `6702` with 40 episodes, do you think it makes sense to allow this script to update `6702` to 25+50+10 episodes?

Reply:
I think 85 is more correct than 45 (did you miscalculate?)

There are still issues with this approach when episodes are missing in previous seasons, but I can't think of any way to handle that.

Reply:
(Yes, I did)

Agree that 85 to me is more correct than 45 even if previous seasons have not been fully watched, I wouldn't understand why someone would skip seasons.

Regarding episodes from previous seasons being missing, doesn't seem really feasible so I guess our users will have to use the other configuration `start` to fix it. But we don't use this `start` here, I think that could be improved.

Reply:
I think I will create a new PR to add that functionality and keep this PR clean for now with the same behaviour